### PR TITLE
MM-35127 CRT: Marking thread as unread doesn't set the timestamp correctly

### PR DESCRIPTION
**Pull request by reflog**
#### Summary

pass correct timestamp on 'mark as unread'
#### Ticket Link

https://mattermost.atlassian.net/browse/MM-35127
#### Related Pull Requests

- Has server changes (please link here)
https://github.com/mattermost/mattermost-server/pull/17507
- Has redux changes (please link here)
- Has mobile changes (please link here)

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: components/threading/global_threads/thread_pane/thread_pane.tsx [hunk 1]
```diff
@@ -53,9 +54,15 @@ const ThreadPane = ({
...
     const channel = useSelector((state: GlobalState) => getChannel(state, {id: channelId}));
     const post = useSelector((state: GlobalState) => getPost(state, thread.id));
-
+    const postsInThread = useSelector((state: GlobalState) => getPostsForThread(state, {rootId: post.id}));
     const selectHandler = useCallback(() => select(), []);
+    let unreadTimestamp = post.edit_at || post.create_at;
 
+    // if we have the whole thread, get the posts in it, sorted. First post - root post, second post - oldest reply. Use that timestamp
+    if (postsInThread.length > 1) {
+        const p = postsInThread[postsInThread.length - 2];
```
Comment (ashishbhate) [inline components/threading/global_threads/thread_pane/thread_pane.tsx:63]: Why do we do `postsInThread.length - 2` here? Isn't the oldest reply: `postsInThread.length-1`?

Comment (reflog) [reply components/threading/global_threads/thread_pane/thread_pane.tsx:63]: @ashishbhate nope. the `postsInThread.length-1` item is the root post, see the comment above :)

Comment (ashishbhate) [reply components/threading/global_threads/thread_pane/thread_pane.tsx:63]: `postsInThread` is ordered from newest post  to the root post?

The comment is a bit confusing. `First post - root post`: From the word "first", I would assume it means index 0 in the array, so the opposite ordering of what it actually is.

Comment (reflog) [reply components/threading/global_threads/thread_pane/thread_pane.tsx:63]: @ashishbhate yeah, reverse order. i'll fix the comment

**Review by ashishbhate — APPROVED** [review]

**Review by ogi-m — APPROVED** [review]

**Pull request merged**